activesupport/lib/active_support/evented_file_update_checker.rb

```diff
@@ -28,11 +57,12 @@ def initialize(files, dirs = {}, &block)
             raise LoadError, "Could not load the 'listen' gem. Add `gem 'listen'` to the development group of your Gemfile", e.backtrace
           end
         end
-        Listen.to(*dtw, &method(:changed)).start
       end
+      boot!
     end
 
     def updated?
+      boot! unless @pid_hash[Process.pid]
```

Comment:
In practice it's fine if we accidentally boot more than one listener per process, we want to make sure we boot at least once though. `Concurrent::Hash` doesn't have a `compute_if_absent` locking method